test/Java10andUp/src/org/openj9/test/java/lang/management/ThreadMXBean/APITestJava10.java

```diff
@@ -0,0 +1,252 @@
+			if (stackDepth > 0) {
+				myMethod(stackDepth - 1);
+			} else { // Create the shortest stack possible.
+				synchronized (waitObject) {
+					try {
+						waitObject.wait();
+					} catch (InterruptedException e) {
+						// ignore
+					}
+				}
+			}
+			logger.debug("end TestThread #"+Thread.currentThread().getId());
+		}
+
+		/*
+		 * Build up a stack of specified depth.
+		 */
+		public Integer myMethod(Integer depth) {
+			if (depth > 0) {
+				myMethod(depth - 1);
+			} else {
+				synchronized (waitObject) {
+					try {
+						waitObject.wait();
+					} catch (InterruptedException e) {
+						// ignore
+					}
+				}
+			}
+			return Integer.valueOf(0);
+		}
+	}
+
+	class MethodHandleTestThread extends TestThread {
+		public MethodHandleTestThread(int depth) {
+			super(depth);
+		}
+
+		@Override
+		public void run() {
+			try {
+				Method m = TestThread.class.getMethod("myMethod", Integer.class);
+				m.invoke(this, Integer.valueOf(stackDepth));
+			} catch (NoSuchMethodException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+				fail("Unexpected exception", e);
+			}
+		}
+	}
+
+	class LambdaTestThread extends TestThread {
+		public LambdaTestThread(int depth) {
+			super(depth);
+		}
+
+		@Override
+		public void run() {
+			try {
+				UnaryOperator<Integer> myLambda = s -> myMethod(s);
+				myLambda.apply(Integer.valueOf(stackDepth));
+			} catch (IllegalArgumentException e) {
+				fail("Unexpected exception", e);
+			}
+		}
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testDumpLimits() throws InterruptedException {
+		logger.debug("testDumpLimits");
+		ArrayList<Thread> myThreads = new ArrayList<>();
+		Thread currentThread = Thread.currentThread();
+		long myId = currentThread.getId();
+		myThreads.add(currentThread);
+		long ids[] = new long[MAXDEPTH];
+		HashMap<Long, Integer> expectedDepths = new HashMap<>();
+		for (int requestedDepth = 0; requestedDepth < MAXDEPTH; requestedDepth++) {
+			Thread t = new TestThread(requestedDepth);
+			myThreads.add(t);
+			long id = t.getId();
+			logger.debug("start "+id);
+			t.start();
+			while (t.getState() != Thread.State.WAITING) {
+				try {
+					Thread.sleep(1);
+				} catch (InterruptedException e) {
+					// ignore
+				}
+			}
+			ids[requestedDepth] = id;
+			/*
+			 * every thread will have one stack frame for the run() method,
+			 * plus the specified number of extra frames.
+			 */
+			expectedDepths.put(id, requestedDepth + 1);
+		}
+
+		for (int maxDepth = 0; maxDepth < MAXDEPTH; maxDepth++) {
+			logger.debug("maxDepth = "+maxDepth);
+			for (ThreadInfo ti: myBean.dumpAllThreads(false, false, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				Integer requestedDepthTemp = expectedDepths.get(id);
+				if (Objects.isNull(requestedDepthTemp)) {
+					/* non-test thread */
+					continue;
+				}
+				int requestedDepth = requestedDepthTemp;
+				if (id != myId) {
```

Comment:
Won't this conditional always be true, since myId is not added to expectedDepths?

Reply:
It wasn't always true, but a bug fix made it a tautology.  Refactored and fixed.

Reply:
I now:
a) check all threads against the maximum depth at the top of the loop.
b) omit the check against myId.

Reply:
I still see the check against myId at L178.

Reply:
Right.  Missed that.